Polyhedron/demo/Polyhedron/Plugins/PMP/Orient_soup_plugin.cpp

```diff
@@ -271,5 +295,139 @@ void Polyhedron_demo_orient_soup_plugin::displayNonManifoldEdges()
     QApplication::restoreOverrideCursor();
   }
 }
+void Polyhedron_demo_orient_soup_plugin::createPointsAndPolyline()
+{
+  
+  const CGAL::Three::Scene_interface::Item_id index = scene->mainSelectionIndex();
+  
+  Scene_polygon_soup_item* item =
+    qobject_cast<Scene_polygon_soup_item*>(scene->item(index));
+
+  if(item)
+  {
+    QApplication::setOverrideCursor(Qt::WaitCursor);
+    Scene_points_with_normal_item* points = new Scene_points_with_normal_item();
+    std::vector<std::size_t> nm_vertices;
+    getNMPoints(nm_vertices, item);
+    if(nm_vertices.empty())
+    {
+      delete points;
+      messages->information(tr("There is no non-manifold vertex in this soup."));
+    }
+    else
+    {
+      
+      BOOST_FOREACH(std::size_t id, nm_vertices)
+      {
+        points->point_set()->insert(item->points()[id]);
+      }
+      points->setName(QString("Non Manifold Vertices of %1").arg(item->name()));
+          points->setColor(QColor(Qt::red));
+          
+          scene->addItem(points);
+    }
+    Polygon_soup::Edges nm_edges = item->non_manifold_edges();
+    if(!nm_edges.empty())
+    {
+      Scene_polylines_item* poly = 
+          new Scene_polylines_item();
+      BOOST_FOREACH(Polygon_soup::Edge edge, nm_edges)
+      {
+        Point_3 a(item->points()[edge[0]]), b(item->points()[edge[1]]);
+        Scene_polylines_item::Polyline new_edge;
+        new_edge.push_back(a);
+        new_edge.push_back(b);
+        poly->polylines.push_back(new_edge);
+      }
+      poly->setName(QString("Non Manifold Edges of %1").arg(item->name()));
+      poly->setColor(QColor(Qt::red));
+      
+      scene->addItem(poly);
+    }
+    else
+    {
+      messages->information("Non non-manifold edges were found.");
```

Comment:
Non -> no

And similarly, I would use a dialog. Actually, there should just be one dialog when no edges _nor_ vertices are found (otherwise, users can see the new item created and can understand the output: the dialog is useful when nothing happens, to let them know that some computation _did_ happen and that nothing was found).